src/fondant/pipeline/compiler.py

```diff
@@ -639,19 +639,24 @@ def _check_ecr_pull_through_rule(self) -> None:
     def _patch_uri(self, og_uri: str) -> str:
         full_ref, tag = og_uri.split(":")
 
-        ref, *_ = full_ref.split("/")
+        ref, *repo = full_ref.split("/")
 
-        if ref == "fndnt":
-            logging.info("Reusable component detected, patching URI")
-            # force pullthrough cache to be used
+        def pull_through(repository_name):
             _ = self.ecr_client.batch_get_image(
-                repositoryName=f"{self.ecr_namespace}/{full_ref}",
+                repositoryName=repository_name,
                 imageIds=[{"imageTag": tag}],
             )
             repo_response = self.ecr_client.describe_repositories(
-                repositoryNames=[f"{self.ecr_namespace}/{full_ref}"],
+                repositoryNames=[repository_name],
             )
-            uri = repo_response["repositories"][0]["repositoryUri"] + ":" + tag
+            return repo_response["repositories"][0]["repositoryUri"] + ":" + tag
+
+        if ref == "fndnt":
+            logging.info("Reusable component detected, patching URI")
+            uri = pull_through(f"{self.ecr_namespace}/{full_ref}")
+        elif ref == "public.ecr.aws":
```

Comment:
Maybe we should start to collect constant string values at one place.

Reply:
Agree with defining reused string values, however this one is only used once.